### PR TITLE
Compatibility for Psr/Log 3.x

### PR DESCRIPTION
**Pull request by syehan**
Hi, I just continuing to escalate this issue as same as the issue of #76. and I just going out to resolve the issue as maintainer expected. This PR, by doing linting and unit test under github actions, seems no issue for PHP <= 8.0 .

### CODE REVIEW

File: src/Events/Span.php [hunk 1]
```diff
@@ -101,11 +101,11 @@ public function start(float $startTime = null) // TODO separate out timing from
     /**
      * Stop the Timer
      *
-     * @param integer|null $duration
+     * @param float|null $duration
      *
      * @return void
      */
-    public function stop(int $duration = null) // TODO separate out timing from Event
```
Comment (dstepe) [inline src/Events/Span.php:108]: What is the purpose for the change from integer to float here? The span duration is given in milliseconds, and I believe that is an integer value. We aren't getting fractional milliseconds. Is there a situation you've found that is different?

**Review by dstepe — COMMENTED** [review]
Thanks for the contribution. I'm all for adding proper return types. One change appears to be inconsistent with psr/log compatibility, though, as noted in the comment.

File: src/Events/Span.php (hunk 1 above)
Comment (syehan) [reply src/Events/Span.php:108]: Tbh, i have no clue actually.. seems I just following the changes on the last issue. Probably stand on the integer still be the best option for declaration. No Issue for that :)

**Review by dstepe — APPROVED** [review]
Looks good to me now. Thanks!
